File: FE/src/app/shared/models/view-models/all-game-infos-view-model.ts
import { PlayerViewModel } from 'src/app/shared/models/view-models/player-view-model';
import { SessionState } from './../../constants/session-states';
import { GameCardViewModel } from './game-card-model';
import { RulesViewModel } from './rules-view-model';
export class AllGameInfosViewModel {
    public rules?: RulesViewModel;
    public players: Array<PlayerViewModel> = [];
    public currentPlayer?: PlayerViewModel;
    public currentCard?: GameCardViewModel;
    public isReverseDirection: boolean = false;

    public sessionCreator?: PlayerViewModel;
    public sessionState: SessionState = SessionState.Play;

    public myCards: Array<GameCardViewModel> = [];
}

File: BE/Game/Models/ViewModels/AllGameInfosViewModel.cs
using Game.Entities;
using Game.Enums;

namespace Game.Models.ViewModels {
  public class AllGameInfosViewModel {
    public RulesViewModel Rules { get; set; }
    public List<PlayerViewModel> Players { get; set; }
    public PlayerViewModel CurrentPlayer { get; set; }
    public CardViewModel CurrentCard { get; set; }
    public bool IsReverseDirection { get; set; }

    public PlayerViewModel SessionCreator { get; set; }
    public SessionState SessionState { get; set; }

    public List<CardViewModel> MyCards { get; set; }

    public AllGameInfosViewModel(Session session, Player mePlayer) {
      Rules = new RulesViewModel(session.Rules);
      Players = session.Players.Select(x => new PlayerViewModel(x)).ToList();
      CurrentPlayer = new PlayerViewModel(session.CurrentPlayer);
      CurrentCard = new CardViewModel(session.CurrentCard);
      IsReverseDirection = session.IsReversing;
      SessionState = session.State;
      SessionCreator = new PlayerViewModel(session.Creator);

      MyCards = mePlayer.Cards.Select(x => new CardViewModel(x)).ToList();
    }
  }
}
